miro_preprocessor/miro_image_to_dynamo/main.tf

```diff
@@ -0,0 +1,46 @@
+module "miro_image_to_dynamo" {
+  source = "../../terraform/lambda"
+  source_dir = "${path.module}/target"
+  description = "Push image json into DynamoDB"
+  name = "miro_image_to_dynamo"
+  alarm_topic_arn = "${var.lambda_error_alarm_arn}"
+  environment_variables = {
+    MIRO_S3_BUCKET = "${var.s3_bucket_name}"
+    TABLE_NAME = "${var.miro_data_table_name}"
+  }
+}
+
+module "miro_image_to_dynamo" {
+  source = "../../terraform/lambda/trigger_sns"
+  lambda_function_name = "${module.miro_image_to_dynamo.function_name}"
+  sns_trigger_arn = "${var.miro_image_to_dynamo_topic_arn}"
+  lambda_function_arn = "${module.miro_image_to_dynamo.arn}"
+}
+
+resource "aws_iam_role_policy" "miro_image_to_dynamo" {
+  name   = "miro_image_to_dynamo_policy"
+  role   = "${module.miro_image_to_dynamo.role_name}"
+  policy = "${data.aws_iam_policy_document.allow_miro_data_put.json}"
+}
+
+data "aws_iam_policy_document" "allow_miro_data_put" {
+  statement {
+    actions = [
+      "dynamodb:PutItem",
+    ]
+
+    resources = [
+      "${var.miro_data_table_arn}",
+    ]
+  }
+
+  statement {
```

Comment:
I don't think you need this. You can pass an existing policy document down from the root.

Reply:
Yeah, but there is no policy document that matches the permissions that this lambda needs, I still have to create it. I think I prefer if it lives in the same file as the lambda module that uses it (unless it's a generic thing that can be reused) so that we don't have to follow too many links to see what is going on